metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -146,48 +151,72 @@ def build_plan(
 
         return DataflowPlan(plan_id=plan_id, sink_output_nodes=[sink_node])
 
-    def _build_metrics_output_node(self, query_spec: MetricFlowQuerySpec) -> ComputedMetricsOutput[SqlDataSetT]:
-        """Build a node that computes all requested metrics along with linkables."""
-
+    def _build_metrics_output_node(
+        self,
+        metric_specs: Sequence[MetricSpec],
+        queried_linkable_specs: LinkableSpecSet,
+        where_constraint: Optional[SpecWhereClauseConstraint] = None,
+        time_range_constraint: Optional[TimeRangeConstraint] = None,
+    ) -> ComputedMetricsOutput[SqlDataSetT]:
         compute_metrics_nodes: List[ComputedMetricsOutput[SqlDataSetT]] = []
-        for metric_spec in query_spec.metric_specs:
+        for metric_spec in metric_specs:
             logger.info(f"Generating compute metrics node for {metric_spec}")
-            metric = self._metric_semantics.get_metric(metric_spec)
-            metric_input_measure_specs = self._metric_semantics.measures_for_metric(metric_spec)
-
-            logger.info(
-                f"For {metric_spec}, needed measures are:\n"
-                f"{pformat_big_objects(metric_input_measure_specs=metric_input_measure_specs)}"
-            )
+            blank_metric_spec = MetricSpec.from_element_name(metric_spec.element_name)
```

Comment:
This is a bit of a special case where in `MetricSemantics` we use `MetricSpec` as a reference key to obtain things in the container, but due to adding the new fields it ends up not being able to find the matching key. 

ie., `MetricSpec(name="m1", alias="test")` won't match as in the `MetricSemantic` it is only defined as `MetricSpec(name="m1")`.

I originally wanted to refactor it to use `MeasureReferences` instead of `MetricSpec` as reference keys, but it ended up being more work that is outside of this PR's scope. To get around this, I ended up creating this "blank" MetricSpec to use as the key to access the `MetricSemantics`.

Reply:
> I originally wanted to refactor it to use MeasureReferences instead of MetricSpec as reference keys

I could help with that if needed.

Reply:
it's not blocking this PR so can be done at anytime, i could take this on as well if you don't bandwidth.

Reply:
pushed the refactor and made the appropriate changes on this PR